### ISSUE TITLE
eval returns int(1) in situations when it should return null

### ISSUE THREAD

**Issue by dcarrith** [facebook/hhvm#2708]
First, some background on the issue.  I posted two issues to Drupal 7 core and the Rules module to ask why they are using `print eval('?>' . $code);` instead of just `eval('?>' . $code);`.

Shown in context, they are basically doing this:

``` php
ob_start();
print eval('?>' . $code);
$output = ob_get_contents();
```

When running on HHVM, the two places where issues surfaced (for us) are in custom blocks that have PHP code as the content, and in rule actions that contain PHP code with header redirects.  

Here are the two issue write-ups for reference:
https://drupal.org/node/2270509
https://drupal.org/node/2270521

Basically, HHVM is interpreting the code and printing the evaluated code, followed by the number 1 (in the case of the custom block).  In the case of the rules module header redirect, HHVM seems to be appending the number 1 onto the url (which causes it to go to a page that doesn't exist).

Changing `print eval('?>' . $code);` to be just `eval('?>' . $code);` fixes the issue for us.
